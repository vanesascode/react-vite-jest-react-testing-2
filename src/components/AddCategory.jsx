import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  // target is destructured from event (event.target)
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setInputValue("");
    // Sending the input value to the parent component:
    onNewCategory(inputValue.trim());
  };

  return (
    //In this case, the aria-label attribute is used to provide a label for the form, indicating that it is a form (useful for the JEST TEST)
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
