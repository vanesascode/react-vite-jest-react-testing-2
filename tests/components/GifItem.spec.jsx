import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

const title = "title";
const url = "https://one-punch.com/";
const alt = "title";

describe("<GifItem /> tests", () => {
  it("should render correctly", () => {
    const { container } = render(<GifItem title={title} url={url} id="id" />);
    expect(container).toMatchSnapshot();
  });

  it("should show the image with the URL and the ALT indicated", () => {
    render(<GifItem title={title} url={url} id="id" />);

    // screen.debug();

    // console.log(screen.getByRole("img").url);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(alt);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  it("should show the title in the component", () => {
    render(<GifItem title={title} url={url} id="id" />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
