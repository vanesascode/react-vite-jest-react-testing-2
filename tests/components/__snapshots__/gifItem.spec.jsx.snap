// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> tests should render correctly 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://one-punch.com/"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
